### DISCUSSION TITLE
Open Nextcloud AIO Interface <-- should be possible to edit the URL

### DISCUSSION THREAD

**Discussion by wsxws** [nextcloud/all-in-one#2016] [Questions] [answered]
Hi there.
im running AIO-Beta 4.4.1 atm and everthing works fine so far.
Just the button 'Open Nextcloud AIO Interface' within the nextcloud does not have any benefit for me.
The "AIO_URL" is generated automaticly with the AIO-Container and looks like 127.0.0.1:8080 which has 2 problems for me.
1. im running the nextcloud on a webserver, which means that 127.0.0.1 will never fit my needs
2. like the nextcloud the aio is behind a remote proxy as well, so the port 8080 wont also never fit.

i found  a line "AIO_URL": "127.0.0.1:8080" in nextcloud_aio_mastercontainer/_data/data/configuration.json, but if i change that line there it will become reset after AIO restart.

is there any way to change that "AIO_URL" permanently ?

PS
by the way in my docker-compose.yaml it a voulme defined
    volumes:
      - nextcloud_aio_mastercontainer:/mnt/docker-aio-config
but this line is ignored and the data are stored in /var/lib/docker/volumes/nextcloud_aio_mastercontainer instead

**Comment by Zoey2936** [comment]
AIO_URL is set to the url, you used to start the containers, so if you stop the containers over the AIO panel. Open the AIO interface, for example using aio.example.com (when you have a reverse proxy for it), you log in with your password and start the container. Then AIO_URL should be set to aio.exmaple.com

> PS by the way in my docker-compose.yaml it a voulme defined volumes: - nextcloud_aio_mastercontainer:/mnt/docker-aio-config but this line is ignored and the data are stored in /var/lib/docker/volumes/nextcloud_aio_mastercontainer instead

That's correct volumes get stored in /var/lib/docker/volumes/[name] by default

**Comment by wsxws** [comment]
That's correct volumes get stored in /var/lib/docker/volumes/[name] by default  <<-- what the option in the docker-compose is for then ?

**Reply by Zoey2936** [reply]
This can **not** be changed

nextcloud_aio_mastercontainer is volume and docker stores volumes by default in /var/lib/docker/volumes

**Comment by wsxws** [comment]
i stopped the containers
i logged out
opened https://nextaio.wsxws.ws again
logged in
started the containers again
went to admin settings in nextcloud itself
clicked on the 'Open Nextcloud AIO Interface' button
and new window opened with the following url:
https://127.0.0.1:8080/api/auth/getlogin?token=.......

this issue has 2 problems it seems.
1. it seems it uses the address of the remote proxy as url (127.0.0.1)
2. is automaticly adds port 8080 which is the used redirection port but doesnt work behind the proxy

**Reply by Zoey2936** [reply]
Are your reverse proxy headers correctly set?

**Reply by wsxws** [reply]
what would a correctly set proxy heder look like for apache ?
atm it looks like this
```    SSLProxyEngine On
    ProxyPass /.well-known !
    ProxyPass "/" "https://127.0.0.1:8080/"
    ProxyPassReverse "/" "https://127.0.0.1:8080/"
    SSLProxyVerify none
    SSLProxyCheckPeerCN off
    SSLProxyCheckPeerName off
    SSLProxyCheckPeerExpire off
```

**Reply by Zoey2936** [reply]
try adding this: `ProxyPreserveHost On`

**Reply by wsxws** [reply]
that has been a simple thing with big effekt. thanks works fine now. thanks for that

what about the location set in docker-compose.yaml ?
how to change the path /var/lib/docker/volumes/nextcloud_aio-masterconteiner to /mnt/docker-aio-config ?
actual volume settings becomes ignored (like mentioned before)

**Reply by Zoey2936** [reply]
This can should not be changed

**Reply by wsxws** [reply]
sorry, my fault, i did not read properly, everything fine there.
and thanks for your help again

**Comment by Zoey2936** [comment, marked as answer]
Solution: `ProxyPreserveHost On` was missing in the Apache config

**Comment by Docmine17** [comment]
For those who use caddy, try this:

reverse_proxy :port {
**header_up Host {http.request.host}**
}
